### PR TITLE
Remove code related to `cluster/test-deploy`

### PR DESCRIPTION
**Pull request by psalajova**
With the deprecation `cluster/test-deploy` in the release repo, removing related code also in ci-tools repo.

### CODE REVIEW

File: pkg/rehearse/rehearse.go [hunk 1]
```diff
@@ -399,25 +382,6 @@ func determineChangedRegistrySteps(candidate, baseSHA string, logger *logrus.Ent
 	return changedRegistrySteps, nil
 }
 
-func determineChangedClusterProfiles(candidate, baseSHA, headSHA string, prNumber int, configUpdaterCfg prowplugins.ConfigUpdater, logger *logrus.Entry) (*ConfigMaps, error) {
```
Comment (psalajova) [inline pkg/rehearse/rehearse.go:402]: I think we can just delete this function? It is called 'cluster profiles' but it looks like it just deals with config maps, which we no longer need

**Comment by smg247** [comment]
Overall, looks good. Can you also remove the `no-cluster-profiles` option and related code?

**Comment by psalajova** [comment]
I'm not sure if the failure in the `breaking-changes` test doesn't seem related to this PR?

**Pull request merged**